server/plugins/messageStorage/sqlite.ts

```diff
@@ -240,8 +240,8 @@ class SqliteMessageStorage implements ISqliteMessageStorage {
 
 		select += " ORDER BY time DESC LIMIT ? OFFSET ? ";
 		params.push(maxResults.toString());
-		query.offset = parseInt(query.offset as string, 10) || 0;
-		params.push(String(query.offset));
```

Comment:
If you assign `String(query.offset)` to something above we can avoid the `toString/String()/as string`s

Reply:
what do you mean exactly? Not sure I follow.

Reply:
`const offset = String(query.offset)` and use `offset` instead of `query.offset as string` and the rest

Reply:
Probably still need to check for undefined / null (although not if I look a the callers, but you know how it goes) and then it'd be String(query.offset || 0) which doesn't seem to be more readable to me than the above.

Don't have any strong opinions, but why do you prefer the conversion over toString() ?

But I don't know

**A)**
```typescript
const offset = query.offset || 0;
params.push(offset.toString());
```
and **B)**
```typescript
const offset = query.offset || 0;
params.push(String(offset));
```

Does exactly the same, I don't favor one over the other so I picked the same that was already there for the limit clause.

Shall I rewrite it to the B) version?

Reply:
Or we have **C)**
```typescript
const offset = ;
params.push(String(query.offset || 0));
```
Gets a tad busy but avoids the useless temporary var

Reply:
actually... TS prevents us from setting it to not a number doesn't it? So ignore it, I don't need to do the or 0 dance at all